documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,279 @@
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## Format of `login_hint`
+
+This CAMARA document clarifies the values used in login_hint in the following way:
+
+    * **_tel_**
+
+      For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+## Offline Access
+
+#### Refresh Token Issuance
+
+Neither OIDC, CIBA, nor OAuth2 define a way for clients to indicate whether they need a refresh_token. 
+Refresh token issuance is optional and at the discretion of the AZ.
+
+Camara uses the scope `offline_access` in the authorization request to indicate to the AZ that the client requests a refresh token additionally to the access token for Camara API access.
+
+---
+**NOTE**
+
+OIDC defines the scope `offline_access` and it is used to get an access token to be used at the UserInfo endpoint, and also to obtain a request token to refresh this access token.
+
+---
+
+The OIDC rules for `offline_access` as defined in [OIDC Offline Access](https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess) apply. 
+
+The OIDC section on [OIDC Using Refresh Tokens](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens) is not changed by this document.
+
+The OIDC security and privacy considerations regarding offline access and refresh tokens apply e.g. on
+[Token lifetime](https://openid.net/specs/openid-connect-core-1_0.html#TokenLifetime) and [Offline Access Privacy](https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccessPrivacy).
+
+### Refresh Token Usage
+
+This section applies to OIDC Authorization Code Flow and CIBA. Other flows do not need refresh tokens and MUST not use refresh tokens.
+
+In addition to [OIDC Using Refresh Tokens](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens) and [OAuth2 Refreshing an Access Token](https://datatracker.ietf.org/doc/html/rfc6749#section-6) this document defines the following:
+
+     * A new access token MUST not be issued if the user has revoked their consent.
+     * A new access token MUST not be issued if the client status regarding the requested API access has changed.
+
+---
+**NOTE**
+
+New user consent might be required if e.g. the user revoked their consent and the issued access tokens and refresh token are not valid anymore.
```

Comment:
```suggestion

```
Is not relevant for refresh token usage.

Reply:
The user at one point in time gave their consent, but then a week later revokes that consent. Maybe only for part of the originally requested scopes.
Now the client uses the original request token. The AZ "sees" that the consent was revoked and now revokes the refresh token (and all access tokens) and returns invalid_grant. Which the client should interpret as "If I really need that access token, then I should ask for consent again because otherwise I cannot fulfill my contract or I cannot provide that service for my customer"

Reply:
> Now the client uses the original request token. The AZ "sees" that the consent was revoked and now revokes the refresh token (and all access tokens) and returns invalid_grant. 

How can the AZ "see" if the consent was revoked, during the Refresh Token Grant? 

If the client (application) revokes the refresh token when the user revokes their consent, the AZ will be returning an invalid_grant error as outlined in RFC 6749, Section 6 (https://datatracker.ietf.org/doc/html/rfc6749#section-6).

This process ensures that the client cannot act on behalf of the user without their consent. If the client then attempts to generate a new access token using the Authorization Code Flow, the AZ will not automatically grant access. Instead, it will require the client to redirect the user to the consent screen, mirroring the original authorization flow.

Reply:
I think there is no dispute over that the AZ has to check the validity of the refresh token. That validity check in a large telco/aggregator system probably goes to a different host and through some API checkValidityOfRequest(aRequest). Whether the different host is "just" a database of all valid refresh tokens or to a consent-management system, that checks, whether consent for that refresh token was revoked , does probably not make a huge difference. Maybe it does make a difference but that are implementation details.

The security and interoperablity document tries to specify that if the user revoked their consent then the request should return an error instead of an access token.
If the user revoked their consent  week ago, then the refresh token might be invalidated a week ago OR it might be invalidated when the refresh token is used. Either way, the AZ must return invalid_grant.

Reply:
OK, please look at my latest suggestion, which aligns with your comment.